tests/process_monitor/test_process_monitor.py

```diff
@@ -820,3 +823,139 @@ def test_MantarrayProcessesMonitor__sends_two_barcode_poll_commands_to_OKComm_at
     assert mocked_get_dur.call_args_list[1][0][0] == expected_time_1
     assert mocked_get_dur.call_args_list[2][0][0] == expected_time_2
     assert mocked_get_dur.call_args_list[3][0][0] == expected_time_2
+
+
+@pytest.mark.parametrize(
+    "expected_barcode,test_valid,expected_status,test_description",
+    [
+        ("MA200190000", True, BARCODE_VALID_UUID, "stores new valid barcode"),
+        ("M$200190000", False, BARCODE_INVALID_UUID, "stores new invalid barcode"),
+        ("", None, BARCODE_UNREADABLE_UUID, "stores no barcode"),
+    ],
+)
+def test_MantarrayProcessesMonitor__stores_barcode_sent_from_ok_comm__and_no_previously_stored_barcode(
+    expected_barcode,
+    test_valid,
+    expected_status,
+    test_description,
+    test_monitor,
+    test_process_manager,
+):
+    monitor_thread, shared_values_dict, _, _ = test_monitor
+    expected_board_idx = 0
+    test_process_manager.create_processes()
+    from_ok_comm_queue = test_process_manager.queue_container().get_communication_queue_from_ok_comm_to_main(
+        0
+    )
+
+    barcode_comm = {
+        "communication_type": "barcode_comm",
+        "barcode": expected_barcode,
+        "board_idx": expected_board_idx,
+    }
+    if test_valid is not None:
+        barcode_comm["valid"] = test_valid
+    from_ok_comm_queue.put(barcode_comm)
+    assert is_queue_eventually_not_empty(from_ok_comm_queue) is True
```

Comment:
prefer `confirm_queue_is_eventually_of_size` if you know the expected size. the error messages are more helpful

Reply:
or this may be better with the combined function `put_object_into_queue_and_confirm_queue_is_eventually_not_empty`

Reply:
both from new updates I made to `stdlib_utils` and used widely in the revamped handling/testing of queues in the desktop app